### PR TITLE
should not be able to send an empty order

### PR DESCRIPTION
**Pull request by jabraok**

### CODE REVIEW

**Comment by brancusi** [comment]
I think that we should disable the UI unless the order is valid instead of silently returning. This will be a better clue to the user.

I think we should add an `isValid` to the order model and disable if not `isValid` using the [truth helpers](https://github.com/jmurphyau/ember-truth-helpers) lib already installed.

**Comment by jabraok** [comment]
Good review,
Thanks for your solutions. I updated the code and got latest code from development branch.
Please review again

File: app/models/order.js [hunk 1]
```diff
@@ -61,5 +61,10 @@ export default Model.extend(LocationHashable, {
   totalPrice(orderItems, shipping) {
     const orderItemsTotal = orderItems.reduce((acc, cur) => acc + Number(cur.get("quantity") * cur.get("unitPrice")), 0);
     return orderItemsTotal + shipping;
+  },
+
+  @computed("orderItems.[]")
+  isValid(orderItems) {
+    return orderItems.get("length") !== 0;
```
Comment (brancusi) [inline app/models/order.js]: Looks good. I forgot to mention that we can use the `gt` helper to make this more compact: http://emberjs.com/api/classes/Ember.computed.html#method_gt

```js
isValid: gt("orderItems.length", 0)
```

Or you can also use the notEmpty: http://emberjs.com/api/classes/Ember.computed.html#method_notEmpty

```js
isValid: notEmpty("orderItems")
```

**Comment by jabraok** [comment]
It's nice Aram,
From now, I will prefer use the Ember helpers then we code.
Thanks

**Pull request merged**